### PR TITLE
Implements discover action, uses genesis client gem

### PR DESCRIPTION
**Pull request by dalehamel**
requires https://github.com/Shopify/genesis_client/pull/1

@dwradcliffe for review

This moves us to using genesis_client, and implements the discover_ipmi_ip handler.

I'm not sure if updating the address in this way will work, but maybe?

### CODE REVIEW

File: scrappystrap.rb [hunk 1]
```diff
@@ -90,12 +99,30 @@ def handle_ipmi_identify(item)
     device_update(item['device']['sku'], ipmi: { identify_light_status: output.include?('on') })
   end
 
+  def handle_discover_ipmi_ip(item)
+    ipmi_mac = item['device']['ipmi_mac']
+
+    lease = `lease_for_mac #{ipmi_mac}`.strip
```
Comment (dwradcliffe) [inline scrappystrap.rb]: Does this need full path?

File: scrappystrap.rb [hunk 2]
```diff
@@ -90,12 +99,30 @@ def handle_ipmi_identify(item)
...
+
+    begin
+      IPAddr.new(lease)
+    rescue IPAddr::InvalidAddressError
+      bail(item, "Couldn't determine a valid lease for #{ipmi_mac}, got #{lease}")
+    end
+
+    log(item, "Determined IPMI IP address to be #{lease}")
+    device_update(item['device']['sku'], ipmi: { address: address })
```
Comment (dwradcliffe) [inline scrappystrap.rb]: Right now we need to send `mac`, `gateway`, `address`, `netmask` all together. We can probably change this in Genesis though.

**Comment by dalehamel** [comment]
@dwradcliffe now with metaprogramming!

**Comment by dalehamel** [comment]
cc @dturn

**Pull request merged**